src/LLMProviders/chainManager.ts

```diff
@@ -124,14 +124,35 @@ export default class ChainManager {
       // Must update the chatModel for chain because ChainFactory always
       // retrieves the old chain without the chatModel change if it exists!
       // Create a new chain with the new chatModel
-      this.setChain(getChainType());
+      this.setChain(getChainType(), {
+        prompt: this.getEffectivePrompt(customModel),
+      });
       console.log(`Setting model to ${newModelKey}`);
     } catch (error) {
       console.error("createChainWithNewModel failed: ", error);
       console.log("modelKey:", newModelKey);
     }
   }
 
+  private getEffectivePrompt(customModel: CustomModel): ChatPromptTemplate {
```

Comment:
Code is duplicated with other parts of this file. Please reorganize to be able to merge.